File: clojure/swing.clj
(import '(javax.swing JFrame JLabel))

(defn hello []
  (let [frame (JFrame. "Hello World Swing")
        label (JLabel. "Hello World")]
    (doto frame
      (.add label)
      (.setDefaultCloseOperation JFrame/EXIT_ON_CLOSE)
      (.setSize 300 300)
      (.setVisible true))))

(hello)

File: js/swing.js
importPackage(javax.swing);

var frame = new JFrame("Hello World Swing");
frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

var label = new JLabel("Hello World");
frame.getContentPane().add(label);

frame.pack();
frame.setSize(300, 100);
frame.setVisible(true);